### PR TITLE
[GPU] Disable oneDNN post-op Prelu in FC,gemm

### PR DESCRIPTION
**Pull request by hyunback**
oneDNN fc(ip) and gemm does not support pot-op Prelu, so disable fusing it until implement it.

### Tickets:
 - *108233*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 1]
```diff
@@ -704,12 +705,16 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
             if (_lo.get_optimization_attributes().use_onednn_impls) {
                 if (input.is_type<reshape>() || input.is_type<concatenation>())
                     return;
-
+                if (activation_func == cldnn::activation_func::relu_negative_slope &&
+                    (input.is_type<fully_connected>() || input.is_type<gemm>())) {
+                    // prelu fusion is not implemented in oneDNN3.1 (https://jira.devtools.intel.com/browse/MFDNN-9898)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:710]: This is internal URL. Please just use ticket id only.

Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp]: applied

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 2]
```diff
@@ -704,12 +705,16 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
             if (_lo.get_optimization_attributes().use_onednn_impls) {
                 if (input.is_type<reshape>() || input.is_type<concatenation>())
                     return;
-
+                if (activation_func == cldnn::activation_func::relu_negative_slope &&
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:708]: AFAIK, `relu_negative_slope` includes both leaky relu(where slope is a scalar) and prelu(where slope is a vector). Does onednn not support both cases?

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 3]
```diff
@@ -704,12 +705,16 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
...
+                    (input.is_type<fully_connected>() || input.is_type<gemm>())) {
+                    // prelu fusion is not implemented in oneDNN3.1 (CVS-108233)
+                    return;
+                }
                 // Activation should not be fused if oneDNN does NOT support it
                 if (_lo.is_primitive_implemented_for_onednn(input))  {
                     #ifdef ENABLE_ONEDNN_FOR_GPU
                     try {
-                        onednn::convert_activation_func(activation_node.get_primitive()->activation_function);
+                        onednn::convert_activation_func(activation_func);
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:717]: (random spot) could you add a unit test?

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 4]
```diff
@@ -704,12 +705,16 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
             if (_lo.get_optimization_attributes().use_onednn_impls) {
                 if (input.is_type<reshape>() || input.is_type<concatenation>())
                     return;
-
+                if (activation_func == cldnn::activation_func::relu_negative_slope &&
+                    (input.is_type<fully_connected>() || input.is_type<gemm>())) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:709]: this is a black-list way, but you said `prelu` is supported for conv only. Then shouldn't we use white-list for `conv`, instead?

**Review by isanghao — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp (hunk 2 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:708]: No, my expectation is only for Prelu with vector, I''ll update it.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp (hunk 3 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:717]: Do you mean check the segfaul unit-test, correct?

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp (hunk 4 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:709]: Only Conv is not an official, just my idea after reviewed the onednn code.
I think to use white-list is good after okay from onednn(they agree only conv can support it), or test it in our side.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp (hunk 3 above)
Comment (isanghao) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:717]: yes, segfault before this PR, OK after this PR.

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp (hunk 2 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp]: Applied

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp (hunk 3 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:718]: Applied

File: src/plugins/intel_gpu/tests/fusions/fully_connected_fusion_test.cpp [hunk 5]
```diff
@@ -617,4 +617,44 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, fc_fp16_eltwise_sum, ::testing::ValuesIn(s
...
+    auto p = GetParam();
+    create_topologies(
+        input_layout("input", get_input_layout(p)),
+        data("weights", get_mem(get_weights_layout(p))),
+        data("bias", get_mem(get_bias_layout(p))),
+        data("data", get_mem(get_bias_layout(p))),
+        fully_connected("fc_prim", input_info("input"), "weights", "bias", padding(), get_output_dim_size(p), get_input_weights_rank(p)),
+        activation("activation", input_info("fc_prim"), "data", activation_func::relu_negative_slope),
+        reorder("reorder_bfyx", input_info("activation"), p.default_format, data_types::f32)
+    );
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/fully_connected_fusion_test.cpp:631]: why is it fp32 test, not fp16 test?

**Comment by isanghao** [comment]
no perf regression --this ../log/pr/2023-04-24--16-06.report --ref ../log/2023-04-24--19-24.report

**Review by isanghao — APPROVED** [review]

**Pull request merged**